### DISCUSSION TITLE
Improve docs for Immer and Valtio

### DISCUSSION THREAD

**Discussion by dmaskasky** [pmndrs/jotai#2171] [Bug report]
I don't believe this example and other examples for jotai-immer and jotai-valtio adequately demonstrates their functionality

## Immer

https://jotai.org/docs/integrations/immer#atomwithimmer

For example, that draft can be mutated. The following example taken from the docs shows the update as  `(c) => (c = c + 1)`. I feel `(draft) => { ++draft.value }` would be more appropriate.

```jsx
import { useAtom } from 'jotai'
import { atomWithImmer } from 'jotai-immer'

const countAtom = atomWithImmer(0)

const Counter = () => {
  const [count] = useAtom(countAtom)
  return <div>count: {count}</div>
}

const Controls = () => {
  const [, setCount] = useAtom(countAtom)
  // setCount === update : (draft: Draft<Value>) => void
  const inc = () => setCount((c) => (c = c + 1))
  return <button onClick={inc}>+1</button>
}
```

## Valtio
https://jotai.org/docs/integrations/immer#atomwithimmer

For example, that this binding is two-way. I feel adding controls to mutate `proxyState` and showcasing how proxyState and stateAtom change together would be good to add here.

```jsx
import { useAtom } from 'jotai'
import { atomWithProxy } from 'jotai-valtio'
import { proxy } from 'valtio/vanilla'

const proxyState = proxy({ count: 0 })
const stateAtom = atomWithProxy(proxyState)
const Counter = () => {
  const [state, setState] = useAtom(stateAtom)

  return (
    <>
      count: {state.count}
      <button
        onClick={() =>
          setState((prev) => ({ ...prev, count: prev.count + 1 }))
        }>
        button
      </button>
    </>
  )
}
```

**Comment by dmaskasky** [comment]
For your consideration: https://github.com/pmndrs/jotai/pull/2172

**Comment by dai-shi** [comment]
If you think we should show more example than `atomWithImmer(0)`, it should be additional examples and it should have multiple values. ex: `atomWithImmer({ firstName: 'foo', lastName: 'bar' })`.

**Reply by dmaskasky** [reply]
The issue I have with using something simple like `0`, is that it doesn't adequately convey the use for immer.

Also `c => c + 1` isn't how we update draft. We can do that already without vanilla atom.

**Reply by dmaskasky** [reply]
We can adjust these examples to use `{ firstName: 'foo', lastName: 'bar' }`, or even pushing to an array.

**Reply by dai-shi** [reply]
Oh my, that was my huge misunderstanding. I thought `atomWithImmer(0)` would somehow work...
Yes, `atomWithImmer({ value: 0 }) should be correct.
And, now I feel like `jotai-immer` doesn't make much sense for Jotai mental model, and I'd like to remove it from the website.
